viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -678,7 +692,9 @@ void CartoFacade::Start() {
         throw VIAM_CARTO_NOT_IN_IO_INITIALIZED_STATE;
     }
     state = CartoFacadeState::STARTED;
-    StartSaveInternalState();
+    if (!config.cloud_story_enabled) {
+        StartSaveInternalState();
+    }
```

Comment:
Moving cloud story enabled logic from `viam_cartographer.go` to `carto_facade.cc` so that proper state changes can be set and checked still

Reply:
why do state changes need to live in carto_facade now? because we hand off the existing map to cartographer here?

Reply:
this was actually a bug before - the Start & stop functions [change the state](https://github.com/viamrobotics/viam-cartographer/blob/db1e677735299d0d4bc6300807386a420ef97cb3/viam-cartographer/src/carto_facade/carto_facade.cc#L680) of the cartofacade object and[ other functions rely on checking that state to work](https://github.com/viamrobotics/viam-cartographer/blob/db1e677735299d0d4bc6300807386a420ef97cb3/viam-cartographer/src/carto_facade/carto_facade.cc#L575).